pkg/codegen/schema_test.go

```diff
@@ -0,0 +1,202 @@
+			},
+			want: "*int",
+		},
+
+		{
+			// if the field(custom-type) is optional, it will NOT be a pointer if
+			// SkipOptionalPointer flag is set to true
+			name: "Set skip optional pointer type for ref type",
+			fields: fields{
+				Schema: Schema{
+					SkipOptionalPointer: true,
+					RefType:             "CustomType",
+					GoType:              "int",
+				},
+				Required: false,
+			},
+			want: "CustomType",
+		},
+
+		// For the following test cases, SkipOptionalPointer flag is false.
+		{
+			name: "When field is required and not nullable",
+			fields: fields{
+				Schema: Schema{
+					SkipOptionalPointer: false,
+					GoType:              "int",
+				},
+				Required: true,
+				Nullable: false,
+			},
+			want: "int",
+		},
+
+		{
+			name: "When field is required and nullable",
+			fields: fields{
+				Schema: Schema{
+					SkipOptionalPointer: false,
+					GoType:              "int",
+				},
+				Required: true,
+				Nullable: true,
+			},
+			want: "*int",
+		},
+
+		{
+			name: "When field is optional and not nullable",
+			fields: fields{
+				Schema: Schema{
+					SkipOptionalPointer: false,
+					GoType:              "int",
+				},
+				Required: false,
+				Nullable: false,
+			},
+			want: "*int",
+		},
+
+		{
+			name: "When field is optional and nullable",
+			fields: fields{
+				Schema: Schema{
+					SkipOptionalPointer: false,
+					GoType:              "int",
+				},
+				Required: false,
+				Nullable: true,
+			},
+			want: "*int",
+		},
+
+		// Following tests cases for non-nullable and required; and skip pointer is not opted
+		{
+			name: "When field is readOnly it will always be pointer",
+			fields: fields{
+				Schema: Schema{
+					SkipOptionalPointer: false,
+					GoType:              "int",
+				},
+				ReadOnly: true,
+				Required: true,
+			},
+			want: "*int",
+		},
+
+		{
+			name: "When field is readOnly and read only pointer disabled",
+			fields: fields{
+				GlobalStateDisableRequiredReadOnlyAsPointer: true,
+				Schema: Schema{
+					SkipOptionalPointer: false,
+					GoType:              "int",
+				},
+				ReadOnly: true,
+				Required: true,
+			},
+			want: "int",
+		},
+
+		{
+			name: "When field is readOnly and optional",
+			fields: fields{
+				Schema: Schema{
+					SkipOptionalPointer: false,
+					GoType:              "int",
+				},
+				ReadOnly: true,
+				Required: false,
+			},
+			want: "*int",
+		},
+		{
+			name: "When field is readOnly and optional and and read only pointer disabled",
+			fields: fields{
+				GlobalStateDisableRequiredReadOnlyAsPointer: true,
+				Schema: Schema{
+					SkipOptionalPointer: false,
+					GoType:              "int",
+				},
+				ReadOnly: true,
+				Required: false,
+			},
+			want: "*int",
+		},
+
+		// When field is write only, it will always be pointer unless pointer is
+		// skipped by setting SkipOptionalPointer flag
+		{
+			name: "When field is write only",
+			fields: fields{
+				GlobalStateDisableRequiredReadOnlyAsPointer: true,
```

Comment:
are we missing a case for the case where this is set to `false`?

Reply:
added...